nixos/modules/services/x11/desktop-managers/gnome.nix

```diff
@@ -59,7 +59,7 @@ let
     enableGnomePanel = true;
   } ++ cfg.flashback.customSessions;
 
-  notExcluded = pkg: mkDefault (!(lib.elem pkg config.environment.gnome.excludePackages));
+  notExcluded = pkg: mkDefault (!(lib.elem (lib.getName pkg) (map lib.getName config.environment.gnome.excludePackages)));
```

Comment:
getName is too much magic for my taste. (Yuck parseDrvName.) Maybe just uwe the pname attribute?

Reply:
`getName` is what the other logic uses, and the point of it is that you can't assume all derivations have a `pname`. I'm not sure why the existing logic doesn't just use the `name` field though.